### PR TITLE
Updated Tuxedo family

### PR DESCRIPTION
**Pull request by DongHan2016**
1. Move the current OracleTuxedo to OracleTuxedo/OracleCore
2. Copy OracleTSAM to OracleTuxedo/TSAMPlus
3. Added following dockerfiles
a. ART_SALT_TMA
b. PerformancePack
c. SALT_TMA
d. TuxedoFull
e. TuxedoJolt
f. TuxedoMessageQ

### CODE REVIEW

**Comment by DongHan2016** [comment]
The folder OracleTSAM should be removed from https://github.com/oracle/docker-images after the PR is approved.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COPY install_tmasna.sh tmasna12.2.2.rsp tmasna122200_64_linux_x86_64.zip install_tmatcp.sh tmatcp12.2.2.rsp tmatcp122200_64_linux_x86_64.zip $tuxedo_rp install_tuxedorp.sh /home/oracle/Downloads/
+# Change user to oracle to install TMA
+USER oracle
+# Set working directory
+WORKDIR /home/oracle/Downloads
+# TUXDIR environment variable is needed before installation
+ENV ORACLE_HOME=/home/oracle/tuxHome
+ENV TUXDIR /home/oracle/tuxHome/tuxedo12.2.2.0.0
+# Install Tuxedo RP
+RUN sh install_tuxedorp.sh $tuxedo_rp
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: All of these `RUN` directives should be combined into a single directive.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# TMA SNA needs Tuxedo RP003 before installation.
+ARG tuxedo_rp
+# Copy all the files needed by the installation to the container
+COPY install_tmasna.sh tmasna12.2.2.rsp tmasna122200_64_linux_x86_64.zip install_tmatcp.sh tmatcp12.2.2.rsp tmatcp122200_64_linux_x86_64.zip $tuxedo_rp install_tuxedorp.sh /home/oracle/Downloads/
+# Change user to oracle to install TMA
+USER oracle
+# Set working directory
+WORKDIR /home/oracle/Downloads
+# TUXDIR environment variable is needed before installation
+ENV ORACLE_HOME=/home/oracle/tuxHome
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: These two `ENV` directives should be combined into a single directive.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Use an official Tuxedo runtime as a base image
+FROM oracle/tuxedoartrt:12.2.2
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: Please provide more detailed documentation on how to create or pull this image.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# Install Tuxedo RP
+RUN sh install_tuxedorp.sh $tuxedo_rp
+# Install TMA SNA
+RUN sh install_tmasna.sh
+# Install TMA TCP
+RUN sh install_tmatcp.sh
+# Clean up installer files
+RUN rm -f *.zip install_*.sh tma*.rsp
+# Set working directory
+WORKDIR /home/oracle
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: There is no `ENTRYPOINT` or `CMD` directive. If you create a container from this image, it's not going to do anything.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Introduction
+This folder contains the information and examples of how to use [Tuxedo](http://oracle.com/tuxedo) with [Docker](https://www.docker.com/).
+
+## Contents
+It is based on the WebLogic Server dockerization (is that even a word?) done by Bruno Borges.
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: That's not a word. Please rewrite this.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+## To use
+1. Into an empty directory:
+  1. Download the Tuxedo TMA 12.2.2 Linux 64 bit installer from OTN
+  2. Download TuxedoARTTMA.zip from this github directory
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: It's a GitHub repo, and where is this file?

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  3. Download the latest Tuxedo rolling patch from My Oracle Support, not less than RP003
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: You cannot require MOS access for publicly accessible Dockerfiles. If a user without Oracle support is unable to downloading this rolling patch, we cannot accept this PR.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+2. Unzip TuxedoARTTMA.zip
+3. Execute buildDockerImage.sh -p <RP Name>
+
+You can then start the image in a new container with:
+
+docker run --privileged -d -p 11122:22 -p 18080:8080 -e DISPLAY=bej301699.cn.oracle.com:1.0 -v /tmp/.X11-unix:/tmp/.X11-unix -h arthost --name tuxedoartrttma oracle/tuxedoartrttma:12.2.2 /sbin/init
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: There are a few problems here: 

- We cannot publish a Dockerfile that requires `--privileged` access for a running container. This is against Oracle security policy guidelines. This needs to be removed.
- What is the purpose of the X11 volume mount? Also, how would this work on non-Linux based platforms, e.g. Docker for Windows or Mac?
- You can't run `/sbin/init` as we don't want to support an entire `init` system. You should be relying on an `ENTRYPOINT` or `CMD` directive to actually run the process required.

File: OracleTuxedo/ART_SALT_TMA/samples/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Use an official Tuxedo runtime as a base image
+FROM oracle/tuxedoartrttma:12.2.2
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/samples/Dockerfile]: Where do I get this image from?

File: OracleTuxedo/ART_SALT_TMA/samples/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+# Add TMA cases to container, this tar ball will automatically be extracted by ADD command
+ADD tma_case.tar.gz /home/oracle/
+# Set user to ROOT
+USER root
+# Base image doesn't include ksh, make and x3270 add them, required by TMA sample cases
+RUN yum -y install x3270; yum -y clean all
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/samples/Dockerfile]: Combine the `RUN` directives into a single directive. Also, I'd suggest using `&&` between commands.

File: OracleTuxedo/ART_SALT_TMA/samples/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+# Change owner
+RUN chown oracle:oracle -R /home/oracle/
+# Set user to oracle
+USER oracle
+# Set working directory
+WORKDIR /home/oracle
+# Used by TMA sample cases
+ENV PATH=.:$PATH
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/samples/Dockerfile]: Combine the `ENV` directives.

File: OracleTuxedo/ART_SALT_TMA/samples/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+ENV LOGNAME=oracle
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/samples/Dockerfile]: Where are the `ENTRYPOINT` or `CMD` directive(s)?

File: OracleTuxedo/ART_SALT_TMA/samples/README.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+wasa                                   #Mainframe Host Name
+/home/oracle/tuxHome/tuxedo12.2.2.0.0  #Tuxedo Path
+2. Make sure that gateway for CICS and IMS on z/OS are configured and running properly, which could be referred to edoc.
+
+## How to build
+build.sh
+
+## How to run
+You can start the image in a new container with:
+docker run --privileged -d -p 11122:22 -p 18080:8080 -e DISPLAY=bej301699.cn.oracle.com:1.0 -v /tmp/.X11-unix:/tmp/.X11-unix -h arthost --name tuxedoartrttmasample oracle/tuxedoartrttmasample:12.2.2 /sbin/init
```
Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/samples/README.md]: As above, we cannot accept any Dockerfile that requires `--privileged` access. Nor should you be running `/sbin/init`. This is a container, not a VM.

**Comment by DongHan2016** [comment]
thanks for the comments. we are updating doc and dockerfiles accordingly. I will close the PR and submit a PR for Core.

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile (hunk 4 above)
Comment (toddinpal) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: I think that will likely be typical of Tuxedo related images.  These images would only be the base for customer created images that actually contain their application.  So we perform the product install, but without an application, there isn't anything to run.

Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/12.2.2/Dockerfile]: Please don't respond to closed pull requests. :)

File: OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md (hunk 7 above)
Comment (toddinpal) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: How are we to apply necessary patches?  As far as I know, MOS is the only vehicle for delivering patches.  Does this mean we can only use GA images, which seems pretty limiting?

Comment (Djelibeybi) [inline OracleTuxedo/ART_SALT_TMA/dockerfiles/README.md]: Please stop replying to closed pull requests.

**Pull request closed without merging**